src/hooks/useApproveCallback.ts

```diff
@@ -107,5 +107,7 @@ export function useApproveCallbackFromTrade(trade?: Trade, allowedSlippage = 0)
   )
   const tradeIsV1 = getTradeVersion(trade) === Version.v1
   const v1ExchangeAddress = useV1TradeExchangeAddress(trade)
+  // Approve token call is made with spender either ROUTER_ADDRESS for uniswapv2 (default)
+  // or with exchangeAddress that is fetched from ExchangeFactory.getExchange(tokenAddress)
```

Comment:
Dima, I think we need to override this hook, and not use V1.

Also, lets try not to change anything in the sources of Uni, just the minimum (like imports making them global)

Reply:
This is not the only place where `ROUTER_ADDRESS` is used, but the most obvious one
Agree that we need a different hook, and a lot of other changes, obviously
Still need to override `ROUTER_ADDRESS`

Reply:
Yes, overriding the const I think was good

Just don't modify their sources, and let's make a dummy `useApproveCallback` hook in our custom dir